Editor config

edieblu opened the pull request:
Something like this? I've added editorconfig, eslint and prettier (I haven't actually run it against all the files yes). Let me know what you think. And if you're busy let me know too and I can ask someone else for review :)

jalada commented:
Happy with editorconfig. I'll need a bit more time to check out eslint, prettier & husky. If anyone else has used those before and so can confidently say yay or nay, happy to defer to them.

edieblu commented:
> Happy with editorconfig. I'll need a bit more time to check out eslint, prettier & husky. If anyone else has used those before and so can confidently say yay or nay, happy to defer to them.

Thanks @jalada! @sztupi knows maybe?

jalada commented:
I've ran all the JS through prettier! I've had a look and prettier looks great, and Husky makes sense.

eslint isn't configured to run as part of husky though. Also, eslint and prettier are fighting; eslint, in its current configuration, tries to change all double quotes to single quotes but prettier changes them to double quotes.

You can see this by doing:

```
$ npx eslint --fix "**/*.js"
```

and then checking the git diff.

Maybe we need to configure eslint some more?

edieblu commented:
> I've ran all the JS through prettier! I've had a look and prettier looks great, and Husky makes sense.
> 
> eslint isn't configured to run as part of husky though. Also, eslint and prettier are fighting; eslint, in its current configuration, tries to change all double quotes to single quotes but prettier changes them to double quotes.
> 
> You can see this by doing:
> 
> ```
> $ npx eslint --fix "**/*.js"
> ```
> 
> and then checking the git diff.
> 
> Maybe we need to configure eslint some more?

Hello :) I've updated the settings, the two should now play together nicely. There are some eslint issues that need fixing (I've added the airbnb linter which is quite strict), before I go fixing those, let me know what you think of this setup.

edieblu commented:
Hey @sztupi these are the changes from yesterday's meetup, and lots of formatting with prettier :)
